testutil/ciao-down/download.go

```diff
@@ -88,25 +106,50 @@ func getUbuntu(ctx context.Context, dest io.WriteCloser, cb progressCB) (err err
 
 	buf := make([]byte, 1<<20)
 	_, err = io.CopyBuffer(dest, pr, buf)
-	_ = resp.Body.Close()
 
 	if err == nil && int(pr.downloaded/1000000)%10 != 0 {
-		downloadProgress(progress{downloadedMB: pr.totalMB, totalMB: pr.totalMB})
+		cb(progress{downloadedMB: pr.totalMB, totalMB: pr.totalMB})
 	}
 
 	return
 }
 
-func downloadUbuntu(ctx context.Context, instanceDir string, cb progressCB) (string, error) {
-	imgPath := path.Join(instanceDir, guestImageName)
+func downloadProgress(p progress) {
+	if p.totalMB >= 0 {
+		fmt.Printf("Downloaded %d MB of %d\n", p.downloadedMB, p.totalMB)
+	} else {
+		fmt.Printf("Downloaded %d MB\n", p.downloadedMB)
+	}
+}
+
+func downloadFile(ctx context.Context, URL, ciaoDir string, cb progressCB) (string, error) {
```

Comment:
Yes, assuming we allow concurrent prepares.  But there will be lots of other issues too unrelated to downloaded that will also need to be addressed, for example downloading of the ubuntu base image and modifying the UI to cope with parallel prepares.